Source/portal.cpp

```diff
@@ -10,203 +10,174 @@ int WarpDropY[MAXPORTAL] = { 40, 40, 40, 40 };
 
 void __cdecl InitPortals()
 {
-	int i; // edi
+    int i;
 
-	for(i = 0; i < MAXPORTAL; i++)
-	{
-		if(delta_portal_inited(i))
-			portal[i].open = 0;
-	}
+    for (i = 0; i < MAXPORTAL; i++) {
+        if (delta_portal_inited(i))
+            portal[i].open = FALSE;
+    }
 }
 
-void __fastcall SetPortalStats(int i, int o, int x, int y, int lvl, int lvltype)
+void __fastcall SetPortalStats(int i, BOOL o, int x, int y, int lvl, int lvltype)
 {
-	portal[i].x = x;
-	portal[i].setlvl = 0;
-	portal[i].y = y;
-	portal[i].open = o;
-	portal[i].level = lvl;
-	portal[i].ltype = lvltype;
+    portal[i].x = x;
+    portal[i].setlvl = FALSE;

```

Comment:
Looks like an odd order of assignments, if you move x next to y does it break bin exact?

Reply:
The setlvl assignment can likely go anywhere without changing the assembly, I left it as it was because it was already marked bin exact. Cosmetics can come after everything's correct.

Reply:
I would be nice to get as much done on the first pass so we don't have to revisit things, it's also a good chance that we would miss stuff like this at later points since we won't be going threw things are detailed.